### ISSUE TITLE
create package for DSM 7.0

### ISSUE THREAD

**Issue by nirev** [tailscale/tailscale-synology#35]
DSM 7.0 Beta developer guide:
https://global.download.synology.com/download/Document/Software/DeveloperGuide/Firmware/DSM/7.0/enu/DSM_Developer_Guide_7_0_Beta.pdf

**Comment by bradfitz** [comment]
What's different?

> Any privileged operation should be accomplished via resource worker.

Is this going to break TUN?

**Comment by nirev** [comment]
That's what I'm thinking.

Synology is going to release the beta version this week, I'm hoping I can run it in a VM to test.

**Comment by quexten** [comment]
I'm running DSM 7.0 Beta on a DS 918+. Happy to report that Tailscale works without issues on it.

**Comment by nirev** [comment]
@quexten this is great! I'm both happy and surprised that it worked! I haven't added any privilege configuration, so both the install scripts and tailscaled are running with root under DSM 6.x

Also according to the new developer guide, the path where packages are stored changed.
Can you post the file tree of `/var/packages/tailscale` ?

thank you

**Comment by quexten** [comment]
```
quexten@valhalla:~$ find /var/packages/tailscale/
/var/packages/tailscale/
/var/packages/tailscale/home
/var/packages/tailscale/target
/var/packages/tailscale/var
/var/packages/tailscale/conf
/var/packages/tailscale/conf/resource
/var/packages/tailscale/conf/resource.own
/var/packages/tailscale/INFO
/var/packages/tailscale/enabled
/var/packages/tailscale/etc
/var/packages/tailscale/tmp
/var/packages/tailscale/scripts
/var/packages/tailscale/scripts/postupgrade
/var/packages/tailscale/scripts/preupgrade
/var/packages/tailscale/scripts/installer
/var/packages/tailscale/scripts/start-stop-status
/var/packages/tailscale/scripts/postuninst
/var/packages/tailscale/scripts/postinst
```
All files still belong to the root user & group. I had tailscale installed on DSM 6 and migrated.

**Comment by bradfitz** [comment]
It's probably running in some compatibility mode for a limited time to not break people's exiting packages.

It sounds like Synology won't be allowing DSM 6.x packages into their package center store going forward, though, so we still have to work with the DSM 7.x SDK regardless. I imagine we might need to give up on TUN entirely and just go full userspace netstack. I should finish that change,.

**Comment by nirev** [comment]
Yes, looks like that is the case.
I'm going to try installing DSM 7.0 in a vm and have a look at some packages here:
https://prerelease.synology.com/en-global/download/dsm70_beta?builtin_name=synology_apollolake_218%2B

There's a lot of undocumented things that are done for some of them (the Docker one in particular seems to use all the features possible), so maybe it will be possible to set up TUN in some way.

**Comment by andersthorsen** [comment]
I noticed that there is a bit of work going on with DSM 7 and Wireguard so that might be of help once someone gets time to look at it. 
https://github.com/runfalk/synology-wireguard/pull/71

**Comment by bs** [comment]
Looking forward to this!

**Comment by kjacobsen** [comment]
Just chiming in to report similar install issues on DSM7 Beta on DS920+. Let me know if you need someone to help testing fixes.

**Comment by kjacobsen** [comment]
Latest release seems to be the right format, but borks on the requirement for root permissions.

**Comment by maisem** [comment]
This is fixed as of #61
